guacamole/src/main/webapp/app/navigation/directives/guacUserMenu.js

```diff
@@ -134,6 +135,64 @@ angular.module('navigation').directive('guacUserMenu', [function guacUserMenu()
                 return authenticationService.isAnonymous();
             };
 
+            /**
+             * All available actions for the current user.
+             */
+            $scope.actions = [ ];
+
+            // Initialize fullscreen functions
+            var docElem = document.documentElement;
+            var requestFullscreen = docElem.mozRequestFullScreen
+                || docElem.webkitRequestFullscreen
+                || docElem.msRequestFullscreen;
+            var exitFullscreen = document.mozCancelFullScreen
+                || document.webkitExitFullscreen
+                || document.msExitFullscreen;
+            if (!!requestFullscreen && !!exitFullscreen) {
+                // Bind browser-specific fullscreen functions
+                if (!docElem.requestFullscreen) {
+                    docElem.requestFullscreen = (requestFullscreen).bind(docElem);
+                }
+                if (!document.exitFullscreen) {
+                    document.exitFullscreen = (exitFullscreen).bind(document);
+                }
+    
+                /**
+                 * Toggles fullscreen for the Guacamole page.
+                 */
+                $scope.fullscreen = function fullscreen() {
+                    var fullscreenElem = !!(document.fullscreenElement 
+                        || document.msFullscreenElement
+                        || document.mozFullScreenElement
+                        || document.webkitFullscreenElement);
+    
+                    if (!fullscreenElem) {
+                        document.documentElement.requestFullscreen();
+                    }
+                    else {
+                        document.exitFullscreen();
+                    }
+                };
+    
+                /**
+                 * Action which requests fullscreen for the Guacamole page.
+                 */
+                var FULLSCREEN_ACTION = {
+                    name      : 'USER_MENU.ACTION_FULLSCREEN',
+                    className : 'fullscreen',
+                    callback  : $scope.fullscreen
+                };
```

Comment:
While I agree there is no need for these to be declared except when fullscreen is supported, this `if` block and the controller as a whole might be more readable were `FULLSCREEN_ACTION` and `$scope.fullscreen()` declared elsewhere, ideally grouped logically with `LOGOUT_ACTION` and its callback.